src/stan/mcmc/hmc/nuts/base_nuts.hpp

```diff
@@ -152,9 +151,9 @@ namespace stan {
         values.push_back(this->energy_);
       }
 
-      bool compute_criterion(Eigen::VectorXd& p_sharp_minus,
-                             Eigen::VectorXd& p_sharp_plus,
-                             Eigen::VectorXd& rho) {
+      virtual bool compute_criterion(Eigen::VectorXd& p_sharp_minus,
```

Comment:
This shouldn't be too much overhead, given that there's a dot-product getting called, but I don't like the idea in general of making functions virtual just for testing.  We shouldn't need to do that, but we can talk about that later.  Let's just get this in for now.